pinot-spi/src/main/java/org/apache/pinot/spi/env/PinotConfiguration.java

```diff
@@ -356,7 +356,14 @@ public long getProperty(String name, long defaultValue) {
    * @return the property String value. Fallback to default value if missing.
    */
   public String getProperty(String name, String defaultValue) {
-    Object rawProperty = getRawProperty(name, defaultValue);
+    String relaxedPropertyName = relaxPropertyName(name);
+    if (!_configuration.containsKey(relaxedPropertyName)) {
+      return defaultValue;
+    }
+    Object rawProperty = _configuration.getProperty(relaxedPropertyName);
+    if (CommonsConfigurationUtils.needInterpolate(rawProperty)) {
+      return _configuration.getString(relaxPropertyName(name), defaultValue);
+    }
     if (rawProperty instanceof List) {
       return StringUtils.join(((ArrayList) rawProperty).toArray(), ',');
```

Comment:
Do we need to interpolate here? Or do we need to handle List when reading string?

Reply:
fixed this according to an idea proposed by @dd-willgan in a comment early on. I didn't get it but now I see what's going on with multi-value interpolation. I expanded unit tests for such case. pls review.